### PR TITLE
refactor: Improve ConversationState API for selfConversation

### PR DESCRIPTION
**Pull request by atomrc**
This adds 2 new methods to the conversation state that improves how self conversations are accessed

### CODE REVIEW

File: src/script/conversation/ConversationRepository.ts [hunk 1]
```diff
@@ -2589,8 +2588,7 @@ export class ConversationRepository {
     }
 
     const isBackendEvent = eventData.otr_archived_ref || eventData.otr_muted_ref;
-    const selfConversation = this.conversationState.selfConversation();
-    const inSelfConversation = selfConversation && matchQualifiedIds(selfConversation, conversationId);
+    const inSelfConversation = this.conversationState.isSelfConversation(conversationId);
```
Comment (arjita-mitra) [inline src/script/conversation/ConversationRepository.ts:2591]: Can we rename it? its a little confusing between inSelfConversation and isSelfConversation

**Review by przemvs — APPROVED** [review]

**Review by PatrykBuniX — APPROVED** [review]

File: src/script/conversation/ConversationRepository.ts [hunk 2]
```diff
@@ -2722,11 +2720,11 @@ export class ConversationRepository {
    * @returns Resolves when the event was handled
    */
   private async onMessageHidden(eventJson: MessageHiddenEvent) {
-    const {conversation: conversationId, data: eventData, from} = eventJson;
+    const {conversation, qualified_conversation, data: eventData, from} = eventJson;
```
Comment (arjita-mitra) [inline src/script/conversation/ConversationRepository.ts:2723]: can we use camelCase here for the qualified_conversation to maintain consistency

**Review by V-Gira — APPROVED** [review]

**Review by arjita-mitra — CHANGES_REQUESTED** [review]

File: src/script/conversation/ConversationRepository.ts (hunk 1 above)
Comment (atomrc) [reply src/script/conversation/ConversationRepository.ts:2591]: We want to know if the message received is, actually, _in_ the self conversation (thus the name of the variable). 
What would you suggest as another name?

File: src/script/conversation/ConversationRepository.ts (hunk 2 above)
Comment (atomrc) [reply src/script/conversation/ConversationRepository.ts:2723]: This is something that comes from the backend, we cannot really change it. 
If you look at the next line, it's used to create a camelCase variable. Does it seem fine to you?

Comment (arjita-mitra) [reply src/script/conversation/ConversationRepository.ts:2723]: we could do qualified_conversation: qualifiedConversation, but its okay as we are assigning it to a variable afterwards

File: src/script/conversation/ConversationRepository.ts (hunk 1 above)
Comment (arjita-mitra) [reply src/script/conversation/ConversationRepository.ts:2591]: sounds good then

**Review by arjita-mitra — APPROVED** [review]

**Pull request merged**